### ISSUE TITLE
A11y keyboard focus control in the Grid List Component

### ISSUE THREAD

**Issue by DaveBest99** [angular/material2#3683]
#### Bug, feature request, or proposal:
Bug - Screen reader user cannot distinguish between grid cell content.

#### What is the expected behavior?
All Grid List Items should have a focusable element, and defined by markup structure.

#### What is the current behavior?
The demo displays a simple Grid List with one row and four columns. The Grid List has no focusable element, and there is no content structure to distinguish between Grid Cells.

#### What are the steps to reproduce?
See below.

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: https://goo.gl/DlHd6U

#### What is the use-case or motivation for changing an existing behavior?
Create a good keyboard user experience.

#### Which versions of Angular, Material, OS, browsers are affected?
Using screen reader JAWS17 with browsers Firefox, Internet Explorer 11, and Google Chrome, on Windows10.

#### Is there anything else we should know?
Screen reader user experience follows:

1. Press Tab key to the Dynamic Grid List element in the Tab Panel.
The screen reader skips over the Tab Panel and moves focus to the page footer. A md-grid-list object could use the tabindex=0 to include it in the focus order.

2. Use the arrow keys to move screen reader focus to the Tab Panel Grid List element.
JAWS says "list of 4 items", and braille display shows "list of 4 items".
As the user presses Down Arrow JAWS speaks only the line text, but does not indicate which lines belong to which Grid row and column item. All list items are merged into a single block of text.
The md-grid-list uses role=list without any ARIA properties to define the Grid structure. Also, each md-grid-tile uses role=listitem with an element defined as role=figure (no such ARIA role).

3. The author must manage focus order for screen readers to understand content context.

3.1 For a simple Grid List (only one row or column) a possible solution may be to use the ARIA role=group property. Use a group to form logical collection of items in a widget. when a group is used in the context of list, authors MUST limit its children to listitem elements.

3.2 A more effective solution would be in using the ARIA role=grid and role=gridcell properties.
The grid pattern can be used to group a set of interactive elements, such as links, buttons, or checkboxes. This may be useful for large md-list objects with interactive elements within each md-list-item. ARIA properties and assistive technologies use row and column nomenclature when describing and presenting the logical structure of elements with the grid
role. In a grid that presents tabular data, every cell contains a focusable element or is itself focusable, regardless of whether the cell content is editable or interactive.

3.3 Another possible solution may be in using the HTML Table element with the appropriate ARIA properties.

See:
The Roles Model | Accessible Rich Internet Applications (WAI-ARIA) 1.0
http://w3c.github.io/aria-practices/#grid

WAI-ARIA Authoring Practices 1.1
2.13 Grids : Interactive Tabular Data and Layout Containers
http://w3c.github.io/aria-practices/#h-grid

Demo Angular Material - Component Grid List
https://material.angular.io/components/component/grid-list

**Comment by jelbourn** [comment]
There's a bit of nomenclature mix-up here. The grid-list component doesn't actually correspond to role="grid"; it's actually an alternate presentation of role="list" in a two-dimensional display. Our name for the role="grid" component is "data-table", which is still in-progress. 

As such, I think the component is working as intended.
